src/api/auth.ts

```diff
@@ -0,0 +1,42 @@
+import { basicAPI } from './';
+import { RegistrationAPIProps, LoginAPIProps } from './types';
+
+enum AuthURL {
+  SIGNUP = '/auth/signup',
+  SIGNIN = '/auth/signin',
+  LOGOUT = '/auth/logout',
+}
+
+class AuthAPI {
+  login(props: LoginAPIProps) {
+    const { data, handleError, handleLoading, handleSuccess } = props;
+
+    return basicAPI.post({
+      data,
+      handleError,
+      errorMessage: 'Login error',
+      handleLoading,
+      handleSuccess,
+      url: AuthURL.SIGNIN,
+    });
+  }
+
+  registration(props: RegistrationAPIProps) {
```

Comment:
❗️
Более каноничное название
```suggestion
  signup(props: RegistrationAPIProps) {
```

Reply:
переименовал